tools/internal/mdocspec/md.go

```diff
@@ -0,0 +1,93 @@
+package mdocspec
+
+import (
+	"io"
+
+	"github.com/yuin/goldmark"
+	"github.com/yuin/goldmark/ast"
+	"github.com/yuin/goldmark/extension"
+	"github.com/yuin/goldmark/renderer"
+)
+
+const (
+	// OCSpecHeading is the MarkDown heading that MUST precede the yaml
+	// section containing the OC path and RPC listing.
+	OCSpecHeading = "OpenConfig Path and RPC Coverage"
+	// OCSpecHeadingSearchN is the number of previous siblings in the
+	// MarkDown AST to search for OCSpecHeading in order to establish
+	// whether the yaml block might be the OC path and RPC listing.
+	OCSpecHeadingSearchN = 5
```

Comment:
Is this the depth of the tree to search?

Reply:
The goldmark parser I'm using creates a hierarchy of MD nodes, and in general there can be several sibling nodes in between the OCSpecHeading and the actual yaml block. This specifies the number of previous sibling nodes to search for OCSpecHeading before giving up.

Reply:
Hrm, sorry I still don't follow.   In any case, I expect the yaml block to appear immediately after / within the OCSpecHeading.  There could be 10+  headings in a big or highly parameterized test.  So maybe make this number 20?

Reply:
Thanks for the suggestion, I changed the logic to search from the heading rather than from the YAML block. This way we can just search until the next heading of the same or lower level or the end of the document, which is cleaner since we can remove this constant.